core/src/main/java/org/apache/accumulo/core/clientImpl/ScannerImpl.java

```diff
@@ -152,6 +153,16 @@ public synchronized int getBatchSize() {
   @Override
   public synchronized Iterator<Entry<Key,Value>> iterator() {
     ensureOpen();
+
+    if (getConsistencyLevel() == ConsistencyLevel.IMMEDIATE) {
+      try {
+        String tableName = context.getTableName(tableId);
+        context.requireNotOffline(tableId, tableName);
+      } catch (TableNotFoundException e) {
+        throw new RuntimeException("Table not found", e);
```

Comment:
I think we probably need a generic `UncheckedAccumuloException` so we can avoid the base RTE class and be able to easily distinguish from our RTEs and ones that libraries/user code might throw. `UncheckedIOException` is a good model to follow (though we don't always utilize that when we can; not currently, anyway).

Reply:
Created UncheckedAccumuloException in 5b98e2a